html_css/v2/foundations/css-foundations.md

```diff
@@ -0,0 +1,425 @@
+* If you want multiple elements to have the same style, you would have to copy + paste the same style to each individual element, causing lots of unnecessary repetition and more bloat
+* Any inline CSS will override the other two methods, which can cause unexpected results (while we won't dive into it here, this can actually be taken advantage of)
+
+#### Internal CSS
+
+Similar to the inline method, internal CSS (or embedded CSS) also involves adding the CSS within the HTML file itself. The difference with internal CSS, though, is that you place all of the rules inside of a pair of opening and closing `<style>` tags, which are then placed inside of the opening and closing `<head>` tags of your HTML file:
+
+~~~html
+<head>
+  <style>
+    div {
+      color: white;
+      background-color: black;
+    }
+
+    p {
+      color: red;
+    }
+  </style>
+</head>
+<body>...</body>
+~~~
+
+Let's go over the differences here compared to inline CSS. 
+
+First, this is where we have to declare a selector (the `div` and `p` bits). Then we have opening and closing curly braces after the selector, which create a "declaration block". Finally, we place any declarations inside of the declaration block. If we wanted to add other rules, we would just continue adding them inside of the opening and closing `<style>` tags.
+
+This method can be useful for adding unique styles to a *single page* of a website, but it still isn't what many consider to be the preferred method.
+
+#### External CSS
+
+External CSS, aka the preferred method, involves creating a separate file for the CSS and linking it inside of an HTML's opening and closing `<head>` tags:
+
+~~~html
+<!-- index.html -->
+
+<head>
+  <link rel="stylesheet" href="styles.css">
+</head>
+~~~
+~~~css
+/* styles.css */
+
+div {
+  color: white;
+  background-color: black;
+}
+
+p {
+  color: red;
+}
+~~~
+
+First, we add a self-closing `<link>` element inside of the opening and closing `<head>` tags of the HTML file. The `href` attribute is the location of the CSS file, either an absolute URL or, what you'll be utilizing, a URL relative to the location of the HTML file.
+
+Then inside of the newly created `styles.css` file, you'll notice the syntax is basically the same as the internal CSS method. The difference here being we no longer need those `<style>` tags (in fact, there should be no HTML tags within an external CSS file). `styles.css` is just what we went with as the file name here. You can name the file whatever you want as long as the file type is `.css`, though "style" or "styles" is most commonly used.
+
+Why is this the preferred method?
+
+1. It keeps our HTML and CSS separated, which results in the HTML file being smaller and makes things look cleaner
+2. We only need to edit the CSS in *one* place, which is especially handy for websites with multiple pages that all have similar styles.
+
+### Selectors
```

Comment:
as i mentioned above... it might be nice to go straight to 'selectors' after the intro.  Seems more in line with the first paragraph that mentions syntax details (i.e. css is made up of a selector and a property and a value leads well into "so now let's learn about selectors, and then properties and values)

but i won't put up a fight either way.. starting with how to link CSS to HTML is also valid.